tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -0,0 +1,365 @@
+                **{
+                    "name": "new name",
+                    "description": "taxonomy description",
+                    "enabled": True,
+                    "required": False,
+                },
+            )
+
+    @ddt.data(
+        (False, False, status.HTTP_200_OK),
+        (False, True, status.HTTP_200_OK),
+        (True, False, status.HTTP_403_FORBIDDEN),
+        (True, True, status.HTTP_403_FORBIDDEN),
+    )
+    @ddt.unpack
+    def test_update_taxonomy_system_defined(
+        self, create_value, update_value, expected_status
+    ):
+        taxonomy = Taxonomy.objects.create(
+            name="test system taxonomy", system_defined=create_value
+        )
+        taxonomy.save()
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": taxonomy.pk})
+
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.put(
+            url, {"name": "new name", "system_defined": update_value}, format="json"
+        )
+        assert response.status_code == expected_status
+
+        if status.is_success(expected_status):
+            response = self.client.get(url)
+            assert response.data["system_defined"] == False
+
+    def test_update_taxonomy_404(self):
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": 123123})
+
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.put(url, {"name": "new name"}, format="json")
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_403_FORBIDDEN),
+        ("staff", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_patch_taxonomy(self, user_attr, expected_status):
+        taxonomy = Taxonomy.objects.create(
+            name="test patch taxonomy", enabled=False, required=True
+        )
+        taxonomy.save()
+
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": taxonomy.pk})
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.patch(
+            url, {"name": "new name", "required": False}, format="json"
+        )
+        assert response.status_code == expected_status
+
+        # If we were able to update the taxonomy, check if the name changed
+        if status.is_success(expected_status):
+            response = self.client.get(url)
+            check_taxonomy(
+                response.data,
+                response.data["id"],
+                **{
+                    "name": "new name",
+                    "enabled": False,
+                    "required": False,
+                },
+            )
+
+    @ddt.data(
+        (False, False, status.HTTP_200_OK),
+        (False, True, status.HTTP_200_OK),
+        (True, False, status.HTTP_403_FORBIDDEN),
+        (True, True, status.HTTP_403_FORBIDDEN),
```

Comment:
We allow changing a taxonomy from system_defined=False to system_defined=True, but not vice versa? That seems weird.

If we say that "Even taxonomy admins cannot change system taxonomies." then I don't think anyone should be allowed to create them via the API either.

Reply:
> We allow changing a taxonomy from system_defined=False to system_defined=True, but not vice versa? That seems weird.

It's OK, that's not what this test does.. It shows that `system_defined` is _always ignored_ when passed as part of the taxonomy patch data, so it can't be changed via the REST API.

However, we can make this clearer with [this suggestion](https://github.com/openedx/openedx-learning/pull/63/files#r1275737975).

> If we say that "Even taxonomy admins cannot change system taxonomies." then I don't think anyone should be allowed to create them via the API either.

Yep, that's already true, see [test_create_taxonomy_system_defined](https://github.com/openedx/openedx-learning/pull/63/files#diff-2f11bfdf39bf0bee3b4778c0d84478fa6afd170e57245f804c77b2e9b4d865a4R184-R194).

Reply:
The different status is because for `system_defined=True`, we have a rule (status=403):
https://github.com/openedx/openedx-learning/blob/a7bc37af850cbdc56537d7c8455a050423049708/openedx_tagging/core/tagging/rules.py#L25-L32

For `system_defined=False`, we just drop the value because of the `editable=False` in the model field (status=200):
https://github.com/openedx/openedx-learning/blob/a7bc37af850cbdc56537d7c8455a050423049708/openedx_tagging/core/tagging/models.py#L143-L150

Reply:
Done: 2ccae75d75227729a3beca0b010d0995e4430697